### PR TITLE
Gather per-process CPU and memory metrics.

### LINKED ISSUE #756: Gather per-process CPU and Memory usage
We pretty much already have the code in proc walker.

### LINKED ISSUE #757: Make metrics consistent between nodes
Hosts have % memory usage, containers have it in KB (or MB etc).

We probably want a way of doing both, TBH.

### PR DESCRIPTION
**Pull request by tomwilkie**
Fixes #756, fixes #757

### CODE REVIEW

File: probe/process/walker_darwin.go [hunk 1]
```diff
@@ -92,3 +92,8 @@ func parseLSOF(output string) (map[string]Process, error) {
 	}
 	return processes, nil
 }
+
+// GetDeltaTotalJiffies returns 0 - darwin doesn't have jiffies.
+func GetDeltaTotalJiffies() (uint64, float64, error) {
+	return 0, 0.0, nil
+}
```
Comment (paulbellamy) [inline probe/process/walker_darwin.go]: Won't this cause a flat sparkline? Better to disable the metric.

Comment (paulbellamy) [inline probe/process/walker_darwin.go]: Ah, deltaTotal will be 0, so this won't display.

File: probe/process/walker.go [hunk 2]
```diff
@@ -8,19 +8,22 @@ type Process struct {
...
 // CachingWalker is a walker than caches a copy of the output from another
 // Walker, and then allows other concurrent readers to Walk that copy.
 type CachingWalker struct {
-	cache     []Process
-	cacheLock sync.RWMutex
-	source    Walker
+	cache         []Process
+	previousByPID map[int]Process
+	cacheLock     sync.RWMutex
+	source        Walker
```
Comment (paulbellamy) [inline probe/process/walker.go]: If we don't need guaranteed ordering of `Walk(...)`, it *might* be cleaner to change the type of `cache` to match `previousByPID`, so processes are always indexed by pid.

File: probe/process/reporter.go [hunk 3]
```diff
@@ -59,9 +73,18 @@ func (r *Reporter) processTopology() (report.Topology, error) {
...
 		if p.PPID > 0 {
 			node.Metadata[PPID] = strconv.Itoa(p.PPID)
 		}
+
+		if deltaTotal > 0 {
+			cpuUsage := float64(p.Jiffies-prev.Jiffies) / float64(deltaTotal) * 100.
+			node = node.WithMetric(CPUUsage, report.MakeMetric().Add(now, cpuUsage).WithMax(maxCPU))
+		}
+
+		node = node.WithMetric(MemoryUsage, report.MakeMetric().Add(now, float64(p.RSSBytes)))
```
Comment (paulbellamy) [inline probe/process/reporter.go]: memory usage from this disagrees with top/htop/ps-aux

File: probe/process/walker.go (hunk 2 above)
Comment (tomwilkie) [inline probe/process/walker.go]: We don't rely on it anywhere.  Will give it a go.

File: probe/process/reporter.go (hunk 3 above)
Comment (paulbellamy) [inline probe/process/reporter.go]: Ah, had forgotten to divide ps aux/etc output by 1024 (not 1000).

**Comment by paulbellamy** [comment]
LGTM

**Pull request merged**